vyper/semantics/types/bases.py

```diff
@@ -625,3 +625,6 @@ def __init__(
     def get_signature(self) -> Tuple[Tuple, Optional[BaseTypeDefinition]]:
         new_args, return_type = self.value_type.get_signature()
         return (self.key_type,) + new_args, return_type
+
+    def get_index_type(self):
+        return self.key_type
```

Comment:
this needs to be implemented for every indexable type, just like `validate_index_type()`, otherwise the behavior does not match the name of the function.